packages/loader/driver-utils/src/blobAggregationStorage.ts

```diff
@@ -311,21 +311,24 @@ export class BlobAggregationStorage extends SnapshotExtractor implements IDocume
         if (this.blobCutOffSize === undefined || this.blobCutOffSize < 0) {
             return summary;
         }
-        // Only pack at data store level.
-        const startingLevel = level === 1;
+
+        let shouldCompress: boolean = false;
 
         let aggregator = aggregatorArg;
-        if (startingLevel) {
+        // checking if this is a dataStore tree, since we only pack at data store level
+        if (Object.keys(summary.tree).includes(".component")) {
             assert(aggregator === undefined, 0x0fb /* "logic err with aggregator" */);
+            assert(level === this.packingLevel, "we are not packing at the right level");
             aggregator = new BlobAggregator();
+            shouldCompress = true;
         }
```

Comment:
would be great to have reverse check as well i.e.
} else {
    assert(level !== this.packingLevel);
}